### PR TITLE
translate uniswap v4 whitepaper draft to Chinese

### PR DESCRIPTION
**Pull request by AmazingAng**
translated white paper pdf in whitepaper-v4-draft-zh.pdf

## Related Issue
Which issue does this pull request resolve?

## Description of changes

### CODE REVIEW

**Comment by hensha256** [comment]
Thanks so much for making this translation, thats awesome! Someone from our team is going to review it next week 🙌

**Comment by c1ay** [comment]
Chinese comments need to be removed from the code

**Comment by AmazingAng** [comment]
> Chinese comments need to be removed from the code

Chinese comments in the code are removed.

File: latex/main-zh.tex [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+\section{介绍} \label{sec:introduction}
+\textsc{Uniswap v4}是一种实现在以太坊虚拟机（EVM）上的自动做市商（AMM），用于实现价值的高效交换。与\textsc{Uniswap Protocol}的先前版本一样，\textsc{Uniswap v4}是非托管、不可升级和无需许可的。 \textsc{Uniswap v4}的重点是通过可定制性和gas效率升级的体系结构变化，在\textsc{Uniswap v1}和\textsc{v2}中构建的AMM模型和\textsc{Uniswap v3}中引入的集中流动性模型的基础上进行改进。
+
+\textsc{Uniswap v1} \cite{Adams18}和\textsc{v2} \cite{Adams20}是\textsc{Uniswap Protocol}的前两个版本，分别实现了ERC-20 <> ETH和ERC-20 <> ERC-20的交易，两者都使用了常量乘积做市商（CPMM）模型。 \textsc{Uniswap v3} \cite{Adams21}通过使用在有限价格范围内提供流动性的仓位，提供了更高资本效率的流动性，并引入了多个费用层级。
+
+尽管集中流动性和费用层级增加了流动性提供者的灵活性，并允许实施新的策略，但\textsc{Uniswap v3}不足以支持随着AMM和市场的发展而出现的新功能。
+
+一些功能，例如最初在\textsc{Uniswap v2}中引入的价格预言机，允许集成者利用去中心化的链上定价数据，但这以增加交易者的gas成本为代价，并且对集成者而言不具有可定制性。其他增强功能的想法，包括通过时间加权平均价格做市商（TWAMM） \cite{White2021} 实现的时间加权平均价格订单（TWAP）、波动率预言机、限价订单或动态费用，需要重新实现核心协议，无法由第三方开发者添加到\textsc{Uniswap v3}中。
+
+此外，在以前的\textsc{Uniswap}版本中，部署新的池需要部署新的合约，其成本随着合约字节码的大小而增加。另外，涉及到与多个\textsc{Uniswap}池的交易涉及到多个合约之间的转账和冗余状态更新。自\textsc{Uniswap v2}以来，\textsc{Uniswap}要求ETH被包装成ERC-20，而不是支持原生ETH。这些都带来了gas成本。
```
Comment (antoncoding) [inline latex/main-zh.tex:124]:
```suggestion
此外，在以前的\textsc{Uniswap}版本中，部署新的池需要部署新的合约，其成本随着合约大小而增加。另外，涉及到与多个\textsc{Uniswap}池的交易涉及到多个合约之间的转账和冗余状态更新。自\textsc{Uniswap v2}以来，\textsc{Uniswap}要求ETH被转换成ERC-20，而不是支持原生ETH。这些都增加了gas成本。
```

File: latex/main-zh.tex [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+在\textsc{Uniswap v4}中，我们通过一些值得注意的功能改进了这一点：
+\begin{itemize}
+    \item \emph{Hooks}：\textsc{Uniswap v4}允许任何人使用自定义功能部署新的集中流动性池。对于每个池，创建者可以定义一个“Hooks合约”，该合约在调用的生命周期的关键点执行逻辑。这些Hooks也可以管理池的交换费用以及向流动性提供者收取的提款费用。
+    \item \emph{单例模式}：\textsc{Uniswap v4}摒弃了先前版本中使用的工厂模型，而是实现了一个包含所有池的单个合约。单例模式降低了池的创建成本和多跳交易（multi-hop trade）的成本。
+    \item \emph{闪电记账}：单例使用“闪电记账”机制，要求在锁定结束时池或调用者没有代币欠款。在调用过程中，代币可以用于单例内外的任意数量的操作。通过EIP-1153 \cite{Akhunov2018}中提议的瞬态存储操作码，这种功能将变得高效。闪电记账进一步降低了跨多个池的交易的gas成本，并支持与\textsc{Uniswap v4}的更复杂集成。
```
Comment (antoncoding) [inline latex/main-zh.tex:130]:
```suggestion
    \item \emph{闪电记账}：单例使用“闪电记账”机制，要求在锁定结束时池或调用者没有代币欠款。在调用过程中，代币可以用于单例内外的任意数量的操作。通过EIP-1153 \cite{Akhunov2018}中提议的瞬态存储操作码，这种功能将变得高效。闪电记账进一步降低了跨多个池的交易的gas成本，并支持与\textsc{Uniswap v4}更复杂的整合。
```

File: latex/main-zh.tex [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+    \item \emph{原生ETH}：\textsc{Uniswap v4}恢复了对原生ETH的支持，并支持在\textsc{v4}池中使用原生代币进行配对。ETH交换者和流动性提供者从转账成本更低和去除额外包装成本中受益。
+\end{itemize}
+%% \ref{sec:liquidity-oracle}）。
+
+以下章节详细解释了这些变化和使它们成为可能的体系结构变化。
```
Comment (antoncoding) [inline latex/main-zh.tex:135]:
```suggestion
以下章节详细解释了这些变化，与成就它们的架构变化。
```

File: latex/main-zh.tex [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+一些功能，例如最初在\textsc{Uniswap v2}中引入的价格预言机，允许集成者利用去中心化的链上定价数据，但这以增加交易者的gas成本为代价，并且对集成者而言不具有可定制性。其他增强功能的想法，包括通过时间加权平均价格做市商（TWAMM） \cite{White2021} 实现的时间加权平均价格订单（TWAP）、波动率预言机、限价订单或动态费用，需要重新实现核心协议，无法由第三方开发者添加到\textsc{Uniswap v3}中。
+
+此外，在以前的\textsc{Uniswap}版本中，部署新的池需要部署新的合约，其成本随着合约字节码的大小而增加。另外，涉及到与多个\textsc{Uniswap}池的交易涉及到多个合约之间的转账和冗余状态更新。自\textsc{Uniswap v2}以来，\textsc{Uniswap}要求ETH被包装成ERC-20，而不是支持原生ETH。这些都带来了gas成本。
+
+在\textsc{Uniswap v4}中，我们通过一些值得注意的功能改进了这一点：
+\begin{itemize}
+    \item \emph{Hooks}：\textsc{Uniswap v4}允许任何人使用自定义功能部署新的集中流动性池。对于每个池，创建者可以定义一个“Hooks合约”，该合约在调用的生命周期的关键点执行逻辑。这些Hooks也可以管理池的交换费用以及向流动性提供者收取的提款费用。
+    \item \emph{单例模式}：\textsc{Uniswap v4}摒弃了先前版本中使用的工厂模型，而是实现了一个包含所有池的单个合约。单例模式降低了池的创建成本和多跳交易（multi-hop trade）的成本。
+    \item \emph{闪电记账}：单例使用“闪电记账”机制，要求在锁定结束时池或调用者没有代币欠款。在调用过程中，代币可以用于单例内外的任意数量的操作。通过EIP-1153 \cite{Akhunov2018}中提议的瞬态存储操作码，这种功能将变得高效。闪电记账进一步降低了跨多个池的交易的gas成本，并支持与\textsc{Uniswap v4}的更复杂集成。
+    \item \emph{原生ETH}：\textsc{Uniswap v4}恢复了对原生ETH的支持，并支持在\textsc{v4}池中使用原生代币进行配对。ETH交换者和流动性提供者从转账成本更低和去除额外包装成本中受益。
```
Comment (antoncoding) [inline latex/main-zh.tex:131]:
```suggestion
    \item \emph{原生ETH}：\textsc{Uniswap v4}恢复了对原生ETH的支持，并支持在\textsc{v4}池中使用原生代币进行配对。ETH交换者和流动性提供者从转账成本更低和去除额外转换成本中受益。
```

File: latex/main-zh.tex [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+\end{itemize}
+%% \ref{sec:liquidity-oracle}）。
+
+以下章节详细解释了这些变化和使它们成为可能的体系结构变化。
+
+\section{Hooks} 
+\label{sec:Hooks}
+
+\emph{Hooks}是在池的执行过程中在指定点执行一些开发者定义的逻辑的外部部署合约。这些Hooks允许集成者创建具有灵活和可定制执行的集中流动性池。
```
Comment (antoncoding) [inline latex/main-zh.tex:140]:
```suggestion
\emph{Hooks}是在池的执行过程中在指定点执行一些开发者定义的逻辑的外部部署合约。这些Hooks允许整合方创建具有灵活和可自定义的集中流动性池。
```

**Review by antoncoding — COMMENTED** [review]
Left some suggestions 😄

File: latex/main-zh.tex (hunk 5 above)
Comment (AmazingAng) [reply latex/main-zh.tex:140]: I think 集成者 is better here

File: latex/main-zh.tex (hunk 3 above)
Comment (AmazingAng) [reply latex/main-zh.tex:135]: lgtm

File: latex/main-zh.tex (hunk 4 above)
Comment (AmazingAng) [reply latex/main-zh.tex:131]: I think 包装 is better

**Comment by wanxinwanxin** [comment]
Original translation looks good to me. 
I'm indifferent towards edits suggested by @antoncoding , think it's decent either way.

**Review by ewilz — APPROVED** [review]

**Review by ewilz — CHANGES_REQUESTED** [review]
revoking approval. @aadams says the latex file has errors.....

**Comment by aadams** [comment]
> revoking approval. @aadams says the latex file has errors.....

on further inspection, it was because i was using a different compiler. changed it and now it works. with that one fix request above we can merge!

File: latex/main-zh.tex [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+\item 随波动率变化的动态费用
+\item 流动性提供者的内部化MEV分配机制
+\item 实现中位数、截断或其他自定义预言机
+\end{itemize}
+
+我们预计在未来为特定的Hooks设计撰写独立的白皮书，因为许多Hooks的复杂性与协议本身一样。
+
+\begin{figure*}[ht!]
+    \centering
+    \scalebox{.7225}{
```
[the remark this answers is not in the document]
Comment (AmazingAng) [reply latex/main-zh.tex]: > Can you lower this scalebox value so the pdf is 4 pages instead of 5?

done

**Review by ewilz — APPROVED** [review]

**Pull request merged**